src-test/src/org/openbravo/test/stockReservation/StockReservationTest.java

```diff
@@ -0,0 +1,271 @@
+   */
+  private Reservation findReservationForOrder(Order salesOrder) {
+    OBCriteria<Reservation> reservationCriteria = OBDal.getInstance().createCriteria(Reservation.class);
+    reservationCriteria.add(
+        Restrictions.eq(Reservation.PROPERTY_SALESORDERLINE + ".id", salesOrder.getOrderLineList().get(0).getId()));
+    return (Reservation) reservationCriteria.setMaxResults(1).uniqueResult();
+  }
+
+  /**
+   * Retrieves the list of reservation stock entries for a given reservation.
+   *
+   * @param reservation
+   *     the reservation for which stock entries are to be retrieved
+   * @return a list of {@link ReservationStock} entries associated with the reservation
+   */
+  private List<ReservationStock> findReservationStocksForReservation(Reservation reservation) {
+    OBCriteria<ReservationStock> reservationStockCriteria = OBDal.getInstance().createCriteria(ReservationStock.class);
+    reservationStockCriteria.add(Restrictions.eq(ReservationStock.PROPERTY_RESERVATION + ".id", reservation.getId()));
+    return reservationStockCriteria.list();
+  }
+
+  /**
+   * Creates a sales order for a scenario where inventory is distributed across multiple warehouses.
+   *
+   * @return the created sales order
+   */
+  private Order createOrderMoreThanOneWarehouse() {
+    return createOrder("More than One Warehouse", "DR", "CO", new BigDecimal(2000), StockReservationTestUtils.TAX_ID,
+        "CRP");
+  }
+
+  private Order createOrderManualReservation() {
+    return createOrder("Manual Reservation", "DR", "CO", new BigDecimal(1000), StockReservationTestUtils.TAX_ID, "CR");
+  }
+
+  /**
+   * Creates an inventory count for a specified warehouse and storage bin.
+   *
+   * @param warehouseId
+   *     the ID of the warehouse
+   * @param locatorId
+   *     the ID of the storage bin
+   * @param name
+   *     the name of the inventory count
+   * @return the created inventory count
+   */
+  private InventoryCount createInventoryCountGeneric(String warehouseId, String locatorId, String name) {
+    Locator storageBin = OBDal.getInstance().get(Locator.class, locatorId);
+    ProductPrice productPrice = OBDal.getInstance().get(ProductPrice.class, StockReservationTestUtils.PRODUCT_PRICE);
+    Warehouse warehouse = OBDal.getInstance().get(Warehouse.class, warehouseId);
+
+    return createInventoryCount(name, productPrice, storageBin, warehouse);
+  }
+
+
+  /**
+   * Creates an inventory count for the "RN" warehouse.
+   *
+   * @return the created inventory count
+   */
+  private InventoryCount createInventoryCountRN() {
+    return createInventoryCountGeneric(StockReservationTestUtils.WAREHOUSE_RN_ID,
+        StockReservationTestUtils.LOCATOR_RN_ID, "Warehouse Rn-0-0-0");
+  }
+
+  /**
+   * Creates an inventory count for the "RS" warehouse.
+   *
+   * @return the created inventory count
+   */
+  private InventoryCount createInventoryCountRS() {
```

Comment:
Is it necessary to create the `createInventoryCountRS` and `createInventoryCountRN` methods where it is only a return to another method call?